packages/@aws-cdk/aws-ec2/lib/subnet.ts

```diff
@@ -0,0 +1,88 @@
+import { CidrBlock, NetworkUtils } from './network-util';
+import { ISubnet } from './vpc';
+
+/**
+ * Contains logic which chooses a set of subnets from a larger list, in conjunction
+ * with SubnetSelection, to determine where to place AWS resources such as VPC
+ * endpoints, EC2 instances, etc.
+ */
+export interface ISubnetSelector {
```

Comment:
For IDE discoverability, it seems like these days we are settling on a pattern where this is an `abstract class SubnetSelector` with static factory functions for the various subclasses.

The subclasses don't need to be exported then.

Keep the contract of the class minimal and it's just as good as an interface.

Reply:
How would a user provide their own subnet filter in this case? Would they just extend the `abstract class SubnetFilter` in their own code?

Reply:
Yes